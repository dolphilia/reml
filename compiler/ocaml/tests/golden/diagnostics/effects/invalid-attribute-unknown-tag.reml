@allows_effects(true)
fn demo() -> i64 {
  0
}
