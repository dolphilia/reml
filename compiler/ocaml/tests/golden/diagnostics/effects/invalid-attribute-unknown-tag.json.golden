{
  "diagnostics": [
    {
      "audit": {
        "effect.stage.required": "at_least:stable",
        "effect.stage.actual": "stable",
        "effect.capability": "runtime",
        "effect.attribute.invalid": "attribute",
        "effect.attribute.location": "demo",
        "effect.attribute.reason": "unknown_effect_tag",
        "stage_trace": [
          { "source": "cli_option", "note": "not provided" },
          { "source": "env_var", "note": "REMLC_EFFECT_STAGE not set" },
          { "source": "typer", "stage": "stable" }
        ]
      },
      "extensions": {
        "effects": {
          "diagnostic_payload": {
            "invalid_attributes": [
              {
                "attribute": "@allows_effects(true)",
                "attribute_name": "allows_effects",
                "reason": "unknown_effect_tag",
                "key": null,
                "provided": true,
                "provided_display": "true"
              }
            ],
            "residual_leaks": []
          },
          "invalid_attributes": [
            {
              "attribute": "@allows_effects(true)",
              "attribute_name": "allows_effects",
              "reason": "unknown_effect_tag",
              "key": null,
              "provided": true,
              "provided_display": "true"
            }
          ],
          "stage": { "required": "at_least:stable", "actual": "stable" },
          "provided": true,
          "expected": "効果タグは識別子または文字列で指定してください",
          "attribute": "@allows_effects(true)",
          "stage_trace": [
            { "source": "cli_option", "note": "not provided" },
            { "source": "env_var", "note": "REMLC_EFFECT_STAGE not set" },
            { "source": "typer", "stage": "stable" }
          ],
          "provided_display": "true"
        },
        "effect.stage.required": "at_least:stable",
        "effect.stage.actual": "stable",
        "effect.stage.capability": "runtime",
        "effect.stage.subject": "demo",
        "effect.stage_trace": [
          { "source": "cli_option", "note": "not provided" },
          { "source": "env_var", "note": "REMLC_EFFECT_STAGE not set" },
          { "source": "typer", "stage": "stable" }
        ],
        "effect.invalid_attributes": [
          {
            "attribute": "@allows_effects(true)",
            "attribute_name": "allows_effects",
            "reason": "unknown_effect_tag",
            "key": null,
            "provided": true,
            "provided_display": "true"
          }
        ]
      },
      "notes": [
        "効果タグは識別子または文字列で指定してください",
        "指定された効果タグ: true"
      ],
      "domain": "型システム",
      "code": "effects.syntax.invalid_attribute",
      "severity": "error",
      "message": "効果属性 @allows_effects(true) は無効です",
      "location": {
        "file": "<入力>",
        "line": 1,
        "column": 17,
        "endLine": 1,
        "endColumn": 21
      }
    }
  ]
}
